### PR TITLE
[ARCTIC-1628][Feature][AMS]: Remove the Independent DeleteFiles for the Iceberg Format Table

### PR DESCRIPTION
**Pull request by xujiangfeng001**
## Why are the changes needed?

This is to increase the feature to 0.4.x，for details, please refer to https://github.com/NetEase/arctic/issues/1628.

## Brief change log

Regularly delete iceberg independent deleteFiles by configuring 'clean-independent-delete-files.enabled'.

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes)
  - If yes, how is the feature documented? (docs)

### CODE REVIEW

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/OrphanFilesCleanService.java [hunk 1]
```diff
@@ -198,6 +231,26 @@ private static int clearInternalTableMetadata(UnkeyedTable internalTable, long l
     return deleteFilesCnt;
   }
 
+  private static int clearInternalTableIndependentFiles(UnkeyedTable internalTable) {
+    Set<DeleteFile> independentFiles = UnKeyedTableUtil.getIndependentFiles(internalTable);
+    int cnt = 0;
```
Comment (shidayang) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/OrphanFilesCleanService.java:236]: Maybe don't need this variable， independentFiles.size() is ok

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/OrphanFilesCleanService.java [hunk 2]
```diff
@@ -165,6 +182,22 @@ public static void cleanMetadata(ArcticTable arcticTable, long lastTime) {
...
   }
 
+  public static void cleanIndependentFiles(ArcticTable arcticTable) {
+    int independentFilesCnt = 0;
+    if (arcticTable.isKeyedTable()) {
+      LOG.info("{} start delete independent files from base store", arcticTable.id());
+      independentFilesCnt = clearInternalTableIndependentFiles(arcticTable.asKeyedTable().baseTable());
+      LOG.info("{} total delete {} independent files from base store", arcticTable.id(), independentFilesCnt);
+      LOG.info("{} start delete independent files from change store", arcticTable.id());
+      independentFilesCnt = clearInternalTableIndependentFiles(arcticTable.asKeyedTable().changeTable());
```
Comment (shidayang) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/OrphanFilesCleanService.java:192]: Mixed format change table don't need this feature.

**Review by shidayang — APPROVED** [review]

**Comment by wangtaohz** [comment]
LGTM. 
I can help to add some unit test cases from my local test, then we can merge it as soon as possible.

**Comment by xujiangfeng001** [comment]
> I can help to add some unit test cases from my local test, then we can merge it as soon as possible.

THX, this is really great.

**Pull request merged**